### ISSUE TITLE
Do we need a separate spec for chunking?

### ISSUE THREAD

**Issue by darobin** [darobin/dasl.ing#2]
Chunking can be useful but should be treated separately from CIDs, with its own indirection layer that can encode the required parameters. Is that useful? Should it be part of CAR?

**Comment by 2color** [comment]
The root spec currently has no mention of "larger" data, i.e. files/blobs/directories. While I don't have the answers for all the open questions and debates, I think it's worthy of mention, irrespective of whether you choose to incorporate it.

**Comment by darobin** [comment]
There's BDASL right there in the list of spec on https://dasl.ing/! And https://dasl.ing/bdasl.html!

**Comment by 2color** [comment]
Right! That's useful, but there's no mention of files/directories/UnixFS which means representing anything more than just a blob of data isn't supported.

I think that's worth mentioning since so many of the CIDs in the ecosystem are UnixFS.

**Comment by hannahhoward** [comment]
I think one of the questions I keep wanting to raise that @2color is pointing at is there's a conceptual difference between

- "I had a big chunk of bytes and I'm deciding whether to break it up into multiple blocks for transport"
and
- "I have data with a semantic structure that has meaning to users that I want to break into multiple blocks so users can query only parts of it"

They've been confused cause of the IPFS block limit, but they're not the same.

A Blake3 like algorithm solves the first problem only, not the second.

A directory for example is a simple structural piece of data with selective querying (I just want to follow a path) -- a directory stored as a single blob of bytes I have to download entirely is less useful than a directory I can path into for just what I want. 

Arguably there are ways you can collapse these problems together but they are complicated IMHO (see https://github.com/Gozala/merkle-reference).

**Comment by hannahhoward** [comment]
My argument is once you add "this data will be broken up in interesting ways to enable querying" the "data always hashes to the same CID" proposition becomes somewhat impossible.

**Comment by 2color** [comment]
> "I have data with a semantic structure that has meaning to users that I want to break into multiple blocks so users can query only parts of it"

Cross-linking to https://github.com/darobin/dasl.ing/issues/16#issuecomment-2553381362 which is relevant for this discussion
